src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -0,0 +1,523 @@
+#include <ArborX_DetailsTreeNodeLabeling.hpp>
+#include <ArborX_DetailsUtils.hpp>
+#include <ArborX_LinearBVH.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#if KOKKOS_VERSION >= 30500
+
+namespace ArborX
+{
+namespace Details
+{
+
+struct WeightedEdge
+{
+  int source;
+  int target;
+  float weight;
+
+private:
+  // performs lexicographical comparison by comparing first the weights and then
+  // the unordered pair of vertices
+  friend KOKKOS_FUNCTION constexpr bool operator<(WeightedEdge const &lhs,
+                                                  WeightedEdge const &rhs)
+  {
+    if (lhs.weight != rhs.weight)
+    {
+      return (lhs.weight < rhs.weight);
+    }
+    using KokkosExt::min;
+    auto const lhs_min = min(lhs.source, lhs.target);
+    auto const rhs_min = min(rhs.source, rhs.target);
+    if (lhs_min != rhs_min)
+    {
+      return (lhs_min < rhs_min);
+    }
+    using KokkosExt::max;
+    auto const lhs_max = max(lhs.source, lhs.target);
+    auto const rhs_max = max(rhs.source, rhs.target);
+    return (lhs_max < rhs_max);
+  }
+};
+
+template <class BVH, class Labels, class Edges, class Metric, class Radii>
+struct FindComponentNearestNeighbors
+{
+  BVH _bvh;
+  Labels _labels;
+  Edges _edges;
+  Metric _metric;
+  Radii _radii;
+
+  template <class ExecutionSpace>
+  FindComponentNearestNeighbors(ExecutionSpace const &space, BVH const &bvh,
+                                Labels const &labels, Edges const &edges,
+                                Metric const &metric, Radii const &radii)
+      : _bvh(bvh)
+      , _labels(labels)
+      , _edges(edges)
+      , _metric{metric}
+      , _radii(radii)
+  {
+    auto const n = bvh.size();
+    ARBORX_ASSERT(labels.extent(0) == 2 * n - 1);
+    ARBORX_ASSERT(edges.extent(0) == n);
+    ARBORX_ASSERT(radii.extent(0) == n);
+
+    Kokkos::parallel_for(
+        "ArborX::MST::find_component_nearest_neighbors",
+        Kokkos::RangePolicy<ExecutionSpace>(space, n - 1, 2 * n - 1), *this);
+  }
+
+  KOKKOS_FUNCTION void operator()(int i) const
+  {
+    constexpr int undetermined = -1;
+    constexpr auto inf = KokkosExt::ArithmeticTraits::infinity<float>::value;
+
+    auto const distance = [bounding_volume_i =
+                               HappyTreeFriends::getBoundingVolume(_bvh, i),
+                           &bvh = _bvh](int j) {
+      using Details::distance;
+      return distance(bounding_volume_i,
+                      HappyTreeFriends::getBoundingVolume(bvh, j));
+    };
+
+    auto const component = _labels(i);
+    auto const predicate = [label_i = component, &labels = _labels](int j) {
+      return label_i != labels(j);
+    };
+    auto const leaf_permutation_i =
+        HappyTreeFriends::getLeafPermutationIndex(_bvh, i);
+
+    WeightedEdge current_best{i, undetermined, inf};
+    static_assert(WeightedEdge{undetermined, undetermined, inf} <
+                      WeightedEdge{0, undetermined, inf},
+                  "");
```

Comment:
If you removed the other static assert you should drop this one too

Reply:
I did not notice it because it compiles even with gcc/7.5.0.